src/main/java/edu/umich/its/cpm/Utils.java

```diff
@@ -466,15 +428,35 @@ public static boolean isCurrentUserCPMAdmin(String userId, Environment env) {
 	}
 
 	/*
-	 * get CoSign user
+	 * get CoSign user, for local development user will be passed from url parameter as
+	 * http://localhost:8080/?testUser=<uniqname> together with use.testUser.url from the
+	 * application.properties we will determine valid user. In short this block is mimicking
+	 * cosign user if CoSign is not enabled
 	 */
 	public static String getRemoteUser(HttpServletRequest request, Environment env) {
-		String propertyRemoteUser = env.getProperty(TEST_REMOTEUSER);
+		String user = null;
 		String remoteUser = request.getRemoteUser();
-		if (remoteUser == null || remoteUser.length() == 0) {
-			remoteUser = propertyRemoteUser;		
+		String testUser = request.getParameter(TEST_USER);
+		boolean isTestUrlEnabled = Boolean.valueOf(env.getProperty(ALLOW_USER_URL));
+		String testUserInSession = (String) request.getSession().getAttribute(TEST_USER);
+
+		if (isTestUrlEnabled && testUser != null) {
```

Comment:
Consider reorganizing test order so it can return as soon as the answer is clear.  E.g. if testUrl is enabled and there is a testUser then it will be used and there is no point in going further.